src/lib/comms/sol-http-server-impl-microhttpd.c

```diff
@@ -1034,6 +1034,7 @@ free_request(struct sol_http_request *request)
         switch (param->type) {
         case SOL_HTTP_PARAM_POST_DATA:
             free((char *)param->value.data.filename.data);
+            break;
```

Comment:
After the loop I see no problem, just wondering if inside it we have a double-free scenario.